server/src/test/java/org/elasticsearch/cluster/NodeConnectionsServiceTests.java

```diff
@@ -245,13 +249,111 @@ public String toString() {
         assertConnectedExactlyToNodes(transportService, targetNodes);
     }
 
+    public void testDisconnectionHistory() {
+        final Settings.Builder settings = Settings.builder();
+        settings.put(CLUSTER_NODE_RECONNECT_INTERVAL_SETTING.getKey(), "100ms");
+
+        final DeterministicTaskQueue deterministicTaskQueue = new DeterministicTaskQueue();
+        final ThreadPool threadPool = deterministicTaskQueue.getThreadPool();
+
+        MockTransport transport = new MockTransport(deterministicTaskQueue.getThreadPool());
+        TestTransportService transportService = new TestTransportService(transport, deterministicTaskQueue.getThreadPool());
```

Comment:
Should probably use `threadPool` here too rather than creating distinct threadpools for each service.